docs/types.rst

```diff
@@ -857,13 +859,18 @@ or create a new memory array and copy every element.
         }
     }
 
-.. index:: ! array;literals, !inline;arrays
+.. index:: ! array;literals, ! inline;arrays
+
+Array Literals
+^^^^^^^^^^^^^^
 
-Array Literals / Inline Arrays
-^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+An array literal is a comma-separated list of one or more expressions, enclosed in square
+brackets (``[...]``). For example ``[1, a, f(3)]``. The elements must have a
+type so they can all be implicitly converted to and this is the elementary type of the array.
```

Comment:
Suggestion: There must be a common type all elements can be implicitly converted to. This is the elementary type of the array.